aten/src/ATen/native/native_functions.yaml

```diff
@@ -675,10 +675,13 @@
     CUDA: resize_cuda_
 
 - func: empty_out(Tensor result, IntList size) -> Tensor
+  device_guard: False  # calls resize_ which uses a DeviceGuard
```

Comment:
Eh we should really put device guards manually in native functions, as this heavily depends on their implementation, which is generally far away from here (doesn’t need to happen now, just a general thought) :confused:

Reply:
Yeah, I think we're generally in favor of pushing device guard implementation into kernels. Will have to do it for everything before we can remove the catch-all, though.